### ISSUE TITLE
Servers create behavior can be improved

### ISSUE THREAD

**Issue by rtgoodwin** [jrperritt/rack#177]
Currently, if you try to create a server, and don't specify all required flags, you'll get an output of a single flag missing (example: --name). Fix --name, and you'll get the next missing flag, and so on. 

The create behavior and help text for the command should
1.  indicate the flags that are required at a minimum, i.e. the help command and a blank create command (or incomplete) should remind the user about the required flags, and indicate which are OR pairs (like image-id / image-name, etc.) 

2. and ideally parse out and report all missing flags at once. Example:
rack servers instance create --name --image-id 353532523525235  (hit enter)
"The create command requires additional options you did not specify:
[--flavor-id | --flavor-name]  [--this | --that]
(The UX doesn't have to be exactly that, just a quick example.)

This behavior should probably be considered a model for other multi-flag commands as well. (Whatever model we decide on.)

**Comment by kenperkins** [comment]
I agree that we can be much better on our feedback. I'm comfortable with doing this after public-beta however.
